src/Microsoft.ML.Data/Scorers/WhatTheFeatureScorer.cs

```diff
@@ -0,0 +1,454 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.CommandLine;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using Microsoft.ML.Runtime.Model;
+
+[assembly: LoadableClass(typeof(IDataScorerTransform), typeof(WhatTheFeatureScorerTransform), typeof(WhatTheFeatureScorerTransform.Arguments),
+    typeof(SignatureDataScorer), "WhatTheFeature Scorer", "wtf", "WhatTheFeatureScorer", MetadataUtils.Const.ScoreColumnKind.WhatTheFeature)]
+
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(WhatTheFeatureScorerTransform), typeof(WhatTheFeatureScorerTransform.Arguments),
+    typeof(SignatureBindableMapper), "WhatTheFeature Mapper", "wtf", MetadataUtils.Const.ScoreColumnKind.WhatTheFeature)]
+
+[assembly: LoadableClass(typeof(ISchemaBindableMapper), typeof(WhatTheFeatureScorerTransform), null, typeof(SignatureLoadModel),
+    "WhatTheFeature Mapper", WhatTheFeatureScorerTransform.MapperLoaderSignature)]
+
+namespace Microsoft.ML.Runtime.Data
+{
+    /// <summary>
+    /// The What The Feature scorer is superset of a generic scorer.
+    /// It outputs score columns from Generic Scorer plus for given features provides vector of corresponding feature contributions.
+    /// </summary>
+    public sealed class WhatTheFeatureScorerTransform
```

Comment:
>WhatTheFeatureScorerTransform [](start = 24, length = 29)

Need different name, pls suggest

Reply:
Let's go with `FeatureImportanceCalculationTransform` for now.

Reply:
Not sure if this is a regular Transform. So named it as FeatureImportanceCalculationScorer

---
In reply to: [235164932](https://github.com/dotnet/machinelearning/pull/1677#discussion_r235164932) [](ancestors = 235164932)

Reply:
One last naming change: Please name this with the prefix `FeatureContribution` rather than `FeatureImportance`. This will make it less ambiguous what it's calculating.